implementations/c/ockam/channel/channel_impl.c

```diff
@@ -0,0 +1,297 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+#include "ockam/syslog.h"
+#include "ockam/memory.h"
+#include "ockam/key_agreement.h"
+#include "ockam/key_agreement/xx.h"
+#include "ockam/transport.h"
+#include "ockam/io/impl.h"
+#include "ockam/channel.h"
+#include "channel_impl.h"
+#include "ockam/codec.h"
+
+ockam_memory_t* gp_ockam_channel_memory = NULL;
+
+ockam_error_t channel_read(void*, uint8_t*, size_t, size_t*);
+ockam_error_t channel_write(void*, uint8_t*, size_t);
+
+uint8_t g_encoded_text[MAX_CHANNEL_PACKET_SIZE];
```

Comment:
Not thread-safe

Reply:
Not thread-safe in terms that if we have more than one channel instances, using them from different threads is not safe. Also, using one channel instance from different threads is also unsafe, that may be OK, but at least deserves documenting (as well as all thread-safety/unsafety stuff in the library)